src/core/layout/qgslayoutitemlegend.h

```diff
@@ -49,6 +50,72 @@ class CORE_EXPORT QgsLegendModel : public QgsLayerTreeModel
     QVariant data( const QModelIndex &index, int role ) const override;
 
     Qt::ItemFlags flags( const QModelIndex &index ) const override;
+
+    /**
+     * Set the pointer to the layoutlegend expressioncontext
+     * \param econtext pointer to the context
+     * \since QGIS 3.8
+     */
+    void setLayoutExpContext( QgsExpressionContext *econtext );
+
+    /**
+     * Is used to control QgsLayoutItemLegend::refresh, otherwise segfault may occur when updating featurecount hashes (symbol expressions).
+     * \since QGIS 3.8
+     */
+    bool refreshable = true;
```

Comment:
This variable _might_ be unrequired. I don't know the details about the crash, I'd be happy to see if there's a cleaner method to prevent this crash from happening.

In any case if it _needs_ to stay, it should be private and not public.

Reply:
I agree, its vestigial code, I'll remove it and related functions.